File: June/21-06-2024/Compare two fractions.js
//{ Driver Code Starts

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split("\n").map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        let str = readLine();
        let obj = new Solution();
        let res = obj.compareFrac(str);
        console.log(res);
    }
}

// } Driver Code Ends



class Solution {
    /**
    * @param string str

    * @returns string
    */
    compareFrac(str) {
        // Your Code Start
             // Regular expression to extract fractions
        let fractionRegex = /(-?\d+)\s*\/\s*(-?\d+)/g;
        let matches = [];
        let match;

        // Find all matches
        while ((match = fractionRegex.exec(str)) !== null) {
            matches.push([parseInt(match[1]), parseInt(match[2])]);
        }

        // Extract numerators and denominators
        let a = matches[0][0], b = matches[0][1];
        let c = matches[1][0], d = matches[1][1];

        // Cross-multiplication comparison
        let lhs = a * d;
        let rhs = c * b;

        if (lhs > rhs) {
            return `${a}/${b}`;
        } else if (lhs < rhs) {
            return `${c}/${d}`;
        } else {
            return "equal";
        }
        //Your Code End
    }
}

File: June/21-06-2024/Compare two fractions.py
#User function Template for python3


class Solution:
    def compareFrac (self, str):
#Your Code Start
    
      # Regular expression to extract fractions
        fractionRegex = re.compile(r"(-?\d+)\s*/\s*(-?\d+)")
        matches = fractionRegex.findall(str)
        
        # Parse the fractions
        numerators = [int(match[0]) for match in matches]
        denominators = [int(match[1]) for match in matches]
        
        # Extract numerators and denominators
        a, b = numerators[0], denominators[0]
        c, d = numerators[1], denominators[1]
        
        # Cross-multiplication comparison
        lhs = a * d
        rhs = c * b
        
        if lhs > rhs:
            return "{}/{}".format(a, b)
        elif lhs < rhs:
            return "{}/{}".format(c, d)
        else:
            return "equal"
#Your Code End

#{ 
 # Driver Code Starts
#Initial Template for Python 3
import re

if __name__ == '__main__':
    ob = Solution()
    t = int(input())
    for _ in range(t):
        str = input()
        print(ob.compareFrac(str))

# } Driver Code Ends
